include/common/nmea_parser.h

```diff
@@ -113,29 +113,30 @@ class NMEAParser {
     private:
 
     static std::vector<std::string> parseNMEAstring(const std::string &textToParse,
-                                                    const std::string &NMEA_sentence_id) {
+                                                    const char* NMEA_sentence_id) {
 
         if (textToParse.empty()) throw GPSError("Empty string to parse");
 
         using namespace std;
-        const char delimiter = '$';      // sentences separated by [$]
+        const char delimiter = '\n';      // sentences separated by [$]
         const char w_delimiter = ',';    // elements separated by  [,]
 
         bool found = false;
-        string sentence;
+        string sentence, foundSentence;
         istringstream split(textToParse);
         while (getline(split, sentence, delimiter)) {
-            if (sentence.find(NMEA_sentence_id) != string::npos) {
+            if (std::strstr(sentence.c_str(), NMEA_sentence_id)) {
```

Comment:
:+1: This should be faster

Reply:
why?

Reply:
Hmm, actually maybe not :laughing:. @tenxlenx why did you change ``NMEA_sentence_id`` from a ``std::string``? I assume it was to avoid a type conversion....?

Reply:
This method is correctly finding the string now, the previous one was buggy and did not work properly